azkaban-common/src/main/java/azkaban/project/Project.java

```diff
@@ -39,6 +39,7 @@
   private int version = -1;
   private long createTimestamp;
   private long lastModifiedTimestamp;
+  private String currentUploaderID;
```

Comment:
Basically we want to have a list of projects and users who are allowed to upload to a cluster. 
Seems that your concern is that this authorization is cluster specific and not project specific. And so why do we want to include the uploader information in Project? Yes, you are right. 

For this we need a Cluster Admin who can have a whitelist/blacklist of projects and users who are allowed to upload. Azkaban has users with Admin access and no cluster Admin, so how do we do it?

Project Validation:
We already have a ProjectValidator interface which makes azkaban-plugins to implement project specific validation. So, we can use it to whitelist projects. No changes need in Azkaban for doing this.

User Validation:
We can have a new GenericValidator interface, which takes some specific metadata and validates on it. We pass the current user uploading zip to this and validate it. Also, if we want to validate on specific use-case in the future, if would be easy to add to this metadata.
Does this sound good?